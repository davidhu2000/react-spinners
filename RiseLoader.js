(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'emotion', 'react', 'prop-types', '@emotion/core', 'recompose/onlyUpdateForKeys'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('emotion'), require('react'), require('prop-types'), require('@emotion/core'), require('recompose/onlyUpdateForKeys'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.emotion, global.react, global.propTypes, global.core, global.onlyUpdateForKeys);
    global.RiseLoader = mod.exports;
  }
})(this, function (exports, _emotion, _react, _propTypes, _core, _onlyUpdateForKeys) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _onlyUpdateForKeys2 = _interopRequireDefault(_onlyUpdateForKeys);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  var riseAmount = 30;

  var even = (0, _core.keyframes)('0%{transform:scale(1.1)}25%{translateY(-', riseAmount, 'px)}50%{transform:scale(0.4)}75%{transform:translateY(', riseAmount, 'px)}100%{transform:translateY(0) scale(1.0)}');

  var odd = (0, _core.keyframes)('0%{transform:scale(0.4)}25%{translateY(', riseAmount, 'px)}50%{transform:scale(1.1)}75%{transform:translateY(', -riseAmount, 'px)}100%{transform:translateY(0) scale(0.75)}');

  var Loader = function (_React$Component) {
    _inherits(Loader, _React$Component);

    function Loader() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, Loader);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Loader.__proto__ || Object.getPrototypeOf(Loader)).call.apply(_ref, [this].concat(args))), _this), _this.style = function (i) {
        var _this$props = _this.props,
            color = _this$props.color,
            size = _this$props.size,
            sizeUnit = _this$props.sizeUnit,
            margin = _this$props.margin;


        return (0, _core.css)('{background-color:', color, ';width:', '' + size + sizeUnit, ';height:', '' + size + sizeUnit, ';margin:', '' + margin, ';border-radius:100%;display:inline-block;animation:', i % 2 === 0 ? even : odd, ' 1s 0s infinite cubic-bezier(.15,.46,.9,.6);animation-fill-mode:both;}');
      }, _this.wrapper = function () {
        return _this.props.css || '';
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(Loader, [{
      key: 'render',
      value: function render() {
        var loading = this.props.loading;


        return loading ? _react2.default.createElement(
          'div',
          {
            className: (0, _emotion.css)(this.wrapper())
          },
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(1))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(2))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(3))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(4))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(5))
          })
        ) : null;
      }
    }]);

    return Loader;
  }(_react2.default.Component);

  Loader.propTypes = {
    loading: _propTypes2.default.bool,
    color: _propTypes2.default.string,
    size: _propTypes2.default.number,
    margin: _propTypes2.default.string,
    sizeUnit: _propTypes2.default.string,
    css: _propTypes2.default.string
  };

  Loader.defaultProps = {
    loading: true,
    color: '#000000',
    size: 15,
    margin: '2px',
    sizeUnit: 'px',
    css: ''
  };

  var Component = (0, _onlyUpdateForKeys2.default)(['loading', 'color', 'size', 'margin', 'sizeUnit', 'css'])(Loader);
  Component.defaultProps = Loader.defaultProps;
  exports.default = Component;
});