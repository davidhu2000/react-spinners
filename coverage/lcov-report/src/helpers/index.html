
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    