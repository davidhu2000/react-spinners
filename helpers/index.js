(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});