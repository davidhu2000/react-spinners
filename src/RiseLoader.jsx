import React from 'react';
import PropTypes from 'prop-types';
import { keyframes, css } from 'emotion';
import onlyUpdateForKeys from 'recompose/onlyUpdateForKeys';

const riseAmount = 30;

const even = keyframes`
  0% {transform: scale(1.1)}
  25% {translateY(-${riseAmount}px)}
  50% {transform: scale(0.4)}
  75% {transform: translateY(${riseAmount}px)}
  100% {transform: translateY(0) scale(1.0)}
`;

const odd = keyframes`
  0% {transform: scale(0.4)}
  25% {translateY(${riseAmount}px)}
  50% {transform: scale(1.1)}
  75% {transform: translateY(${-riseAmount}px)}
  100% {transform: translateY(0) scale(0.75)}
`;

class Loader extends React.Component {
    style = i => {
      const {
        color, size, sizeUnit, margin
      } = this.props;

      return css`{
            background-color: ${color};
            width: ${`${size}${sizeUnit}`};
            height: ${`${size}${sizeUnit}`};
            margin: ${`${margin}`};
            border-radius: 100%;
            display: inline-block;
            animation: ${i % 2 === 0 ? even : odd} 1s 0s infinite cubic-bezier(.15,.46,.9,.6);
            animation-fill-mode: both;
        }`;
    };

    wrapper = () => {
      const { className } = this.props;

      return className || '';
    };

    render() {
      const { loading } = this.props;

      return loading ?
        <div className={this.wrapper()}>
          <div className={this.style(1)} />
          <div className={this.style(2)} />
          <div className={this.style(3)} />
          <div className={this.style(4)} />
          <div className={this.style(5)} />
        </div> : null;
    }
}

Loader.propTypes = {
  loading: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.number,
  margin: PropTypes.string,
  sizeUnit: PropTypes.string,
  className: PropTypes.string
};

Loader.defaultProps = {
  loading: true,
  color: '#000000',
  size: 15,
  margin: '2px',
  sizeUnit: 'px',
  className: ''
};

const Component = onlyUpdateForKeys(['loading', 'color', 'size', 'margin', 'sizeUnit', 'className'])(Loader);
Component.defaultProps = Loader.defaultProps;
export default Component;
