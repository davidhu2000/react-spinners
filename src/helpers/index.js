export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
