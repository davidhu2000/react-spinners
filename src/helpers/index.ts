export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
