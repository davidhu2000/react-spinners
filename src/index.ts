import BarLoaderComponent from "./BarLoader";
import BeatLoaderComponent from "./BeatLoader";
import BounceLoaderComponent from "./BounceLoader";
import CircleLoaderComponent from "./CircleLoader";
import ClimbingBoxLoaderComponent from "./ClimbingBoxLoader";
import ClipLoaderComponent from "./ClipLoader";
import ClockLoaderComponent from "./ClockLoader";
import DotLoaderComponent from "./DotLoader";
import FadeLoaderComponent from "./FadeLoader";
import GridLoaderComponent from "./GridLoader";
import HashLoaderComponent from "./HashLoader";
import MoonLoaderComponent from "./MoonLoader";
import PacmanLoaderComponent from "./PacmanLoader";
import PropagateLoaderComponent from "./PropagateLoader";
import PulseLoaderComponent from "./PulseLoader";
import PuffLoaderComponent from "./PuffLoader";
import RingLoaderComponent from "./RingLoader";
import RiseLoaderComponent from "./RiseLoader";
import RotateLoaderComponent from "./RotateLoader";
import ScaleLoaderComponent from "./ScaleLoader";
import SkewLoaderComponent from "./SkewLoader";
import SquareLoaderComponent from "./SquareLoader";
import SyncLoaderComponent from "./SyncLoader";

import {
  LoaderHeightWidthProps,
  LoaderSizeMarginProps,
  LoaderSizeProps,
  LoaderHeightWidthRadiusProps
} from "./interfaces";

export const BarLoader: React.ComponentClass<LoaderHeightWidthProps> = BarLoaderComponent;
export const BeatLoader: React.ComponentClass<LoaderSizeMarginProps> = BeatLoaderComponent;
export const BounceLoader: React.ComponentClass<LoaderSizeProps> = BounceLoaderComponent;
export const CircleLoader: React.ComponentClass<LoaderSizeProps> = CircleLoaderComponent;
export const ClimbingBoxLoader: React.ComponentClass<LoaderSizeProps> = ClimbingBoxLoaderComponent;
export const ClipLoader: React.ComponentClass<LoaderSizeProps> = ClipLoaderComponent;
export const ClockLoader: React.ComponentClass<LoaderSizeProps> = ClockLoaderComponent;
export const DotLoader: React.ComponentClass<LoaderSizeProps> = DotLoaderComponent;
export const FadeLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = FadeLoaderComponent;
export const GridLoader: React.ComponentClass<LoaderSizeMarginProps> = GridLoaderComponent;
export const HashLoader: React.ComponentClass<LoaderSizeProps> = HashLoaderComponent;
export const MoonLoader: React.ComponentClass<LoaderSizeProps> = MoonLoaderComponent;
export const PacmanLoader: React.ComponentClass<LoaderSizeMarginProps> = PacmanLoaderComponent;
export const PropagateLoader: React.ComponentClass<LoaderSizeProps> = PropagateLoaderComponent;
export const PulseLoader: React.ComponentClass<LoaderSizeMarginProps> = PulseLoaderComponent;
export const PuffLoader: React.ComponentClass<LoaderSizeProps> = PuffLoaderComponent;
export const RingLoader: React.ComponentClass<LoaderSizeProps> = RingLoaderComponent;
export const RiseLoader: React.ComponentClass<LoaderSizeMarginProps> = RiseLoaderComponent;
export const RotateLoader: React.ComponentClass<LoaderSizeMarginProps> = RotateLoaderComponent;
export const ScaleLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = ScaleLoaderComponent;
export const SkewLoader: React.ComponentClass<LoaderSizeProps> = SkewLoaderComponent;
export const SquareLoader: React.ComponentClass<LoaderSizeProps> = SquareLoaderComponent;
export const SyncLoader: React.ComponentClass<LoaderSizeProps> = SyncLoaderComponent;
