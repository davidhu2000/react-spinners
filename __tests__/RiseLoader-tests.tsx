import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from "@emotion/jest";
expect.extend(matchers);

import RiseLoader from "../src/RiseLoader";
import { sizeMarginDefaults } from "../src/helpers";
import { commonSpecs, cssSpecs, lengthSpecs, speedMultiplierSpecs } from "./sharedSpecs";

describe("RiseLoader", () => {
  const defaultSize = 15;
  const defaultMargin = 2;
  const defaultColor = "#000000";
  const defaultUnit = "px";
  const defaultSpeed = 1;

  commonSpecs(RiseLoader, sizeMarginDefaults(defaultSize));
  cssSpecs(RiseLoader);

  it("should contain styles created using default props", () => {
    const loader = mount(<RiseLoader />);
    for (let i = 0; i < 5; i++) {
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${defaultSize}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${defaultMargin}${defaultUnit}`);
    }
  });

  it("should render the correct color based on prop", () => {
    const color = "#e2e2e2";
    const loader = mount(<RiseLoader color={color} />);
    for (let i = 0; i < 5; i++) {
      expect(loader.find("span span").at(i)).not.toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", color);
    }
  });

  const sizeExpectStatements = (loader: ReactWrapper, length: number, unit?: string) => {
    for (let i = 0; i < 5; i++) {
      expect(loader.find("span span").at(i)).not.toHaveStyleRule("height", `${defaultSize}${defaultUnit}`);
      expect(loader.find("span span").at(i)).not.toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);

      expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${length}${unit || defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${length}${unit || defaultUnit}`);
    }
  };
  lengthSpecs(RiseLoader, "size", sizeExpectStatements);

  const marginExpectStatements = (loader: ReactWrapper, length: number, unit?: string) => {
    for (let i = 0; i < 5; i++) {
      expect(loader.find("span span").at(i)).not.toHaveStyleRule("margin", `${defaultMargin}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${length}${unit || defaultUnit}`);
    }
  };
  lengthSpecs(RiseLoader, "margin", marginExpectStatements);

  const speedMultiplierExpectStatements = (loader: ReactWrapper, multiplier: number) => {
    for (let i = 0; i < 5; i++) {
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "animation",
        expect.stringContaining(`${defaultSpeed * multiplier}s`)
      );
    }
  };

  speedMultiplierSpecs(RiseLoader, speedMultiplierExpectStatements);
});
