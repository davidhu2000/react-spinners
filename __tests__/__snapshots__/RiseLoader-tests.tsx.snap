// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiseLoader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }

  25% {
    -webkit-transform: translateY(30px);
    -moz-transform: translateY(30px);
    -ms-transform: translateY(30px);
    transform: translateY(30px);
  }

  50% {
    -webkit-transform: scale(1.1);
    -moz-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }

  75% {
    -webkit-transform: translateY(-30px);
    -moz-transform: translateY(-30px);
    -ms-transform: translateY(-30px);
    transform: translateY(-30px);
  }

  100% {
    -webkit-transform: translateY(0) scale(0.75);
    -moz-transform: translateY(0) scale(0.75);
    -ms-transform: translateY(0) scale(0.75);
    transform: translateY(0) scale(0.75);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: scale(1.1);
    -moz-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }

  25% {
    -webkit-transform: translateY(-30px);
    -moz-transform: translateY(-30px);
    -ms-transform: translateY(-30px);
    transform: translateY(-30px);
  }

  50% {
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }

  75% {
    -webkit-transform: translateY(30px);
    -moz-transform: translateY(30px);
    -ms-transform: translateY(30px);
    transform: translateY(30px);
  }

  100% {
    -webkit-transform: translateY(0) scale(1.0);
    -moz-transform: translateY(0) scale(1.0);
    -ms-transform: translateY(0) scale(1.0);
    transform: translateY(0) scale(1.0);
  }
}

.emotion-01yp393e {
  background-color: #000000;
  width: 15px;
  height: 15px;
  margin: 2px;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: animation-0 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6);
  animation: animation-0 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-010df5wf {
  background-color: #000000;
  width: 15px;
  height: 15px;
  margin: 2px;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: animation-1 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6);
  animation: animation-1 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

<Loader
  color="#000000"
  css=""
  loading={true}
  margin={2}
  size={15}
>
  <span
    className="emotion-0"
  >
    <span
      className="emotion-01xdhyk6"
    >
      <span
        className="emotion-01yp393e"
      />
      <span
        className="emotion-010df5wf"
      />
      <span
        className="emotion-01yp393e"
      />
      <span
        className="emotion-010df5wf"
      />
      <span
        className="emotion-01yp393e"
      />
    </span>
  </span>
</Loader>
`;
